Pcap++/src/WinPcapLiveDevice.cpp

```diff
@@ -126,6 +126,36 @@ bool WinPcapLiveDevice::setMinAmountOfDataToCopyFromKernelToApplication(int size
 	return true;
 }
 
+PcapLiveDevice* WinPcapLiveDevice::clone() const
+{
+	WinPcapLiveDevice* retval = nullptr;
+
+	pcap_if_t* interfaceList;
+	char errbuf[PCAP_ERRBUF_SIZE];
+	int err = pcap_findalldevs(&interfaceList, errbuf);
+	if (err < 0)
+	{
+		PCPP_LOG_ERROR("Error searching for devices: " << errbuf);
+		return nullptr;
+	}
+
+	pcap_if_t* currInterface = interfaceList;
+	while (currInterface != nullptr)
+	{
+		if (!strcmp(currInterface->name, getName().c_str()))
+			break;
+		currInterface = currInterface->next;
+	}
+
+	if (currInterface)
+		retval = new WinPcapLiveDevice(currInterface, true, true, true);
+	else
+		PCPP_LOG_ERROR("Can't find interface " << getName().c_str());
+
+	pcap_freealldevs(interfaceList);
+	return retval;
+}
+
```

Comment:
Both `clone()` implementations have mostly all common code. Maybe we can do something like this (it's all pseudo-code that I didn't actually run):
```cpp
class PcapLiveDevice
{
public:
    PcapLiveDevice* clone() const
    {
        // call pcap_findalldevs, iterate over the interfaces and find the right one
        pcap_if_t* currInterface = ...
        if(currInterface)
	    result = cloneInternal(currInterface);
        else
	    PCPP_LOG_ERROR("Can't find interface " << getName().c_str());
        
        // call pcap_freealldevs and do cleanup

        return result;
    }
protected:
    virtual PcapLiveDevice* cloneInternal(void* iface)
    {
        // need to case from `void*` to `pcap_if_t*` to avoid exposing `pcap_if_t*` in the .h file
        return new PcapLiveDevice(iface, true, true, true)
    }
}

class WinPcapLiveDevice : public PcapLiveDevice
{
protected:
    WinPcapLiveDevice* cloneInternal(void* iface) override
    {
        // need to case from `void*` to `pcap_if_t*` to avoid exposing `pcap_if_t*` in the .h file
        return new WinPcapLiveDevice(iface, true, true, true)
    }
}

```

Reply:
Sure, can do that. Btw, `pcap_if_t` is already forward declared in PcapLiveDevice.h anyway, so no need for the void ptr shenanigans.

Reply:
oh, you're right! 👍

Reply:
Implemented 0a037af